Pcap++/header/PcapLiveDevice.h

```
@@ -404,7 +404,7 @@ namespace pcpp
 		/// - Capture is already running
 		/// - Device is not opened
 		/// - Capture thread could not be created
-		virtual bool startCapture(OnPacketArrivesCallback onPacketArrives, void* onPacketArrivesUserCookie);
+		virtual bool startCapture(OnPacketArrivesCallback onPacketArrives, void* onPacketArrivesUserCookie) final;
```

Adding `final` might be a breaking change if someone derived from `PcapLiveDevice`, no? 🤔

---

It will. It will raise a compile time error for overriding a final method. I did that because it is a better behavior than removing the virtual, which would silently continue to work wrongly.

I don't think user derived classes from PcapLiveDevice are really something many would do, nor something we want to support. None of the other device classes seem to support this. The virtuals are mainly to support the overrides of WinPcapDevice.

If we want to support custom behavior, IMO defining extension points is a better way than allowing the entire method to be overridden.

---

I agree we don't necessarily want to support it, but since it's already in the API - do we really want to break it?

---

Eh, we can leave it virtual for now, and have a comment that specified that the `virtual` will be removed in the future?
The reason we had the function as virtual is gone and I think `startCapture` is a bit tightly integrated in the internal logic of `PcapLiveDevice` to allow full override, no?